tests/test_functions/background_trigger/main.py

```diff
@@ -29,8 +29,9 @@ def function(
         data dictionary.
       context (google.cloud.functions.Context): The Cloud Functions event context.
     """
-    filename = event["filename"]
-    value = event["value"]
+    attributes = event.get("attributes", {})
+    filename = event.get("filename", attributes.get("filename"))
+    value = event.get("value", attributes.get("value"))
```

Comment:
This test is taken directly from our internal tests. I think it probably shouldn't be changed unless that internal test has changed as well?

Reply:
Sorry, I'm not sure which internal tests are you referring to? I updated this to accurately reflect the structure of a Pub/Sub event dict, which always looks like this:

```
{
  '@type': 'type.googleapis.com/google.pubsub.v1.PubsubMessage', 
  'attributes': {'hello': 'world'}, // or None
  'data': 'YmVlcCE=' // or empty string
}
```

AFAIK it isn't possible to publish a Pub/Sub message that will invoke a function with arbitrary top level keys in the event dict, which is how this test is currently grabbing `filename` and `value`

Reply:
This one: go/link-to-the-test-for-matt

Reply:
bleh, this is a bit annoying. The problem is that these tests were written to accept [arbitrary payloads](https://github.com/GoogleCloudPlatform/functions-framework-python/blob/master/tests/test_functions.py#L40-L49) that don't necessary match what they will receive in production. Our options are:

1. update the test, as I have done here
2. update the marshalling logic to be more lenient about what it puts in data payload
3. skip adding unit this tests all together

Reply:
went with option 3